sotodlib/mapmaking/ml_mapmaker.py

```diff
@@ -361,9 +373,9 @@ def transeval(self, id, obs, other, map, tod):
 
 class SignalCut(Signal):
     def __init__(self, comm, name="cut", ofmt="{name}_{rank:02}", dtype=np.float32,
-            output=False, cut_type=None):
+            output=False, cut_type=None, glitch_flags:str ="flags.glitch_flags"):
```

Comment:
I would prefer these default to None, as they do in SignalMap.  Then convert None -> flags.glitch_flags, to handle the default.

When you have cascading kwargs, e.g.:
```
  SignalMap(..., glitch_flags=None)
     -> Signal(glitch_flags=glitch_flags)
          ->  self.glitch_flags = glitch_flags
```
it tends to defeat the use of default value declarations for optional parameters.  My practice has been to prefer this format for setting default args:
```
    class Signal:
        def __init__(self, ..., glitch_flags=None):
            if glitch_flags is None:
                glitch_flags = 'flags.glitch_flags'
```
Then subclasses, or whatever, can pass in their default value of None and it propagates cleanly all the way down to whatever base function is comfortable with setting the default value.

However... this isn't necessary relevant if we take glitch_flags out of the base class Signal.  In that case you won't have cascading.  I still think it's a good practice, in code that _might_ become cascade, to use "None" to represent "the default value specified by some lower level".  (Though this makes it harder to use "None" to represent "disable this feature" -- I would normally use False to signal that in cases where it is needed.)

Reply:
Generally, I prefer having the actual default value over a check and set in the code. It makes it very easy know the default value, and it avoids any question as to what is happening. I agree the base class needs to have a default as `None`. Using `kargs` will resolve that and make it much cleaner for sure.